# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/test/gtest/TestQuotaManager.cpp
# Commit: f466599094b7
# Full Hash: f466599094b7e16948794a49b0571e03eb2b4df9
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-06-15 13:49:48
# Regressor Bug: 1962133
# File Overlap Count: 14
# Description:
#   Revert "Bug 1962133 - QM: Add accessed flag to FullOriginMetadata and persist it in the origin metadata file; r=dom-storage-reviewers,hsingh" as requested by dmeehan for causing top crash in Bug 1972221.
#   
#   This reverts commit 433768e64dd71c996694e08005854756b7c45fb3.
# ==============================================================================

diff -r 2408bee8bcc4 -r f466599094b7 dom/quota/test/gtest/TestQuotaManager.cpp
--- a/dom/quota/test/gtest/TestQuotaManager.cpp	Sat Jun 14 05:48:43 2025 +0000
+++ b/dom/quota/test/gtest/TestQuotaManager.cpp	Sun Jun 15 16:46:59 2025 +0300
@@ -416,8 +416,6 @@
 // Test simple OpenClientDirectory behavior and verify that origin access time
 // updates are triggered as expected.
 TEST_F(TestQuotaManager, OpenClientDirectory_Simple) {
-  auto testOriginMetadata = GetTestOriginMetadata();
-
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
 
   ASSERT_NO_FATAL_FAILURE(AssertStorageNotInitialized());
@@ -426,12 +424,6 @@
   const auto saveOriginAccessTimeCountInternalBefore =
       SaveOriginAccessTimeCountInternal();
 
-  // Can't check origin state metadata since storage is not yet initialized.
-
-  auto directoryMetadataHeaderBefore =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_FALSE(directoryMetadataHeaderBefore);
-
   PerformOnBackgroundThread([]() {
     QuotaManager* quotaManager = QuotaManager::Get();
     ASSERT_TRUE(quotaManager);
@@ -464,15 +456,6 @@
                 saveOriginAccessTimeCountInternalBefore,
             2u);
 
-  auto originStateMetadataAfter = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataAfter);
-  ASSERT_TRUE(originStateMetadataAfter->mAccessed);
-
-  auto directoryMetadataHeaderAfter =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_TRUE(directoryMetadataHeaderAfter);
-  ASSERT_TRUE(directoryMetadataHeaderAfter->mAccessed);
-
   ASSERT_NO_FATAL_FAILURE(AssertStorageInitialized());
 
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
@@ -481,8 +464,6 @@
 // Test simple OpenClientDirectory behavior when the origin directory exists,
 // and verify that access time updates are triggered on first and last access.
 TEST_F(TestQuotaManager, OpenClientDirectory_Simple_OriginDirectoryExists) {
-  auto testOriginMetadata = GetTestOriginMetadata();
-
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
 
   ASSERT_NO_FATAL_FAILURE(AssertStorageNotInitialized());
@@ -497,15 +478,6 @@
   const auto saveOriginAccessTimeCountInternalBefore =
       SaveOriginAccessTimeCountInternal();
 
-  auto originStateMetadataBefore = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataBefore);
-  ASSERT_FALSE(originStateMetadataBefore->mAccessed);
-
-  auto directoryMetadataHeaderBefore =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_TRUE(directoryMetadataHeaderBefore);
-  ASSERT_FALSE(directoryMetadataHeaderBefore->mAccessed);
-
   PerformOnBackgroundThread([]() {
     QuotaManager* quotaManager = QuotaManager::Get();
     ASSERT_TRUE(quotaManager);
@@ -538,15 +510,6 @@
                 saveOriginAccessTimeCountInternalBefore,
             2u);
 
-  auto originStateMetadataAfter = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataAfter);
-  ASSERT_TRUE(originStateMetadataAfter->mAccessed);
-
-  auto directoryMetadataHeaderAfter =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_TRUE(directoryMetadataHeaderAfter);
-  ASSERT_TRUE(directoryMetadataHeaderAfter->mAccessed);
-
   ASSERT_NO_FATAL_FAILURE(AssertStorageInitialized());
 
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
@@ -557,8 +520,6 @@
 // solely for updating access time.
 TEST_F(TestQuotaManager,
        OpenClientDirectory_Simple_NonExistingOriginDirectory) {
-  auto testOriginMetadata = GetTestOriginMetadata();
-
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
 
   ASSERT_NO_FATAL_FAILURE(AssertStorageNotInitialized());
@@ -573,14 +534,6 @@
   const auto saveOriginAccessTimeCountInternalBefore =
       SaveOriginAccessTimeCountInternal();
 
-  auto originStateMetadataBefore = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataBefore);
-  ASSERT_FALSE(originStateMetadataBefore->mAccessed);
-
-  auto directoryMetadataHeaderBefore =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_FALSE(directoryMetadataHeaderBefore);
-
   PerformOnBackgroundThread([]() {
     QuotaManager* quotaManager = QuotaManager::Get();
     ASSERT_TRUE(quotaManager);
@@ -618,14 +571,6 @@
                 saveOriginAccessTimeCountInternalBefore,
             0u);
 
-  auto originStateMetadataAfter = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataAfter);
-  ASSERT_TRUE(originStateMetadataAfter->mAccessed);
-
-  auto directoryMetadataHeaderAfter =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_FALSE(directoryMetadataHeaderAfter);
-
   ASSERT_NO_FATAL_FAILURE(AssertStorageInitialized());
 
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
@@ -652,15 +597,6 @@
   const auto saveOriginAccessTimeCountInternalBefore =
       SaveOriginAccessTimeCountInternal();
 
-  auto originStateMetadataBefore = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataBefore);
-  ASSERT_FALSE(originStateMetadataBefore->mAccessed);
-
-  auto directoryMetadataHeaderBefore =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_TRUE(directoryMetadataHeaderBefore);
-  ASSERT_FALSE(directoryMetadataHeaderBefore->mAccessed);
-
   PerformOnBackgroundThread([testOriginMetadata]() {
     QuotaManager* quotaManager = QuotaManager::Get();
     ASSERT_TRUE(quotaManager);
@@ -706,13 +642,6 @@
                 saveOriginAccessTimeCountInternalBefore,
             2u);
 
-  auto originStateMetadataAfter = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_FALSE(originStateMetadataAfter);
-
-  auto directoryMetadataHeaderAfter =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_FALSE(directoryMetadataHeaderAfter);
-
   ASSERT_NO_FATAL_FAILURE(AssertStorageInitialized());
 
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
@@ -723,8 +652,6 @@
 // after the origin access time update triggered by first access has finished,
 // and that access time is updated only on first and last access as expected.
 TEST_F(TestQuotaManager, OpenClientDirectory_Ongoing_OriginDirectoryExists) {
-  auto testOriginMetadata = GetTestOriginMetadata();
-
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
 
   ASSERT_NO_FATAL_FAILURE(AssertStorageNotInitialized());
@@ -739,15 +666,6 @@
   const auto saveOriginAccessTimeCountInternalBefore =
       SaveOriginAccessTimeCountInternal();
 
-  auto originStateMetadataBefore = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataBefore);
-  ASSERT_FALSE(originStateMetadataBefore->mAccessed);
-
-  auto directoryMetadataHeaderBefore =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_TRUE(directoryMetadataHeaderBefore);
-  ASSERT_FALSE(directoryMetadataHeaderBefore->mAccessed);
-
   PerformOnBackgroundThread([saveOriginAccessTimeCountBefore,
                              saveOriginAccessTimeCountInternalBefore]() {
     auto testClientMetadata = GetTestClientMetadata();
@@ -880,15 +798,6 @@
                 saveOriginAccessTimeCountInternalBefore,
             2u);
 
-  auto originStateMetadataAfter = GetOriginStateMetadata(testOriginMetadata);
-  ASSERT_TRUE(originStateMetadataAfter);
-  ASSERT_TRUE(originStateMetadataAfter->mAccessed);
-
-  auto directoryMetadataHeaderAfter =
-      LoadDirectoryMetadataHeader(testOriginMetadata);
-  ASSERT_TRUE(directoryMetadataHeaderAfter);
-  ASSERT_TRUE(directoryMetadataHeaderAfter->mAccessed);
-
   ASSERT_NO_FATAL_FAILURE(AssertStorageInitialized());
 
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
@@ -3213,7 +3122,6 @@
 
   auto originStateMetadata = maybeOriginStateMetadata.extract();
   ASSERT_GT(originStateMetadata.mLastAccessTime, 0);
-  ASSERT_FALSE(originStateMetadata.mAccessed);
   ASSERT_FALSE(originStateMetadata.mPersisted);
 
   ASSERT_NO_FATAL_FAILURE(ShutdownStorage());
